src/core/config/parent_search.h

```diff
@@ -46,18 +46,37 @@
 /**
  * @def OPENTHREAD_CONFIG_PARENT_SEARCH_ENABLE
  *
- * Define as 1 to enable periodic parent search feature.
+ * Define as 1 to enable the periodic parent search feature.
  *
- * When this feature is enabled an end-device/child (while staying attached) will periodically search for a possible
- * better parent and will switch parent if a better one is found.
+ * When this feature is enabled, an end device/child (while staying attached) periodically searches for a potentially
+ * better parent and switches parents if a better one is found.
  *
- * The child will periodically check the average RSS value for the current parent, and only if it is below a specific
- * threshold, a parent search is performed. The `OPENTHREAD_CONFIG_PARENT_SEARCH_CHECK_INTERVAL` specifies the
- * check interval (in seconds) and `OPENTHREAD_CONFIG_PARENT_SEARCH_RSS_THRESHOLD` gives the RSS threshold.
+ * The parent search mechanism depends on whether the device is an FTD child or an MTD child.
  *
- * Since the parent search process can be power consuming (child needs to stays in RX mode to collect parent response)
- * and to limit its impact on battery-powered devices, after a parent search is triggered, the child will not trigger
- * another one before a specified backoff interval specified by `OPENTHREAD_CONFIG_PARENT_SEARCH_BACKOFF_INTERVAL`.
+ * FTD Child
```

Comment:
Should we consider enabling this by default on an FTD child?

Reply:
Yes, that makes sense. I have made the change in a new push and added a comment to encourage enabling this feature on MTD builds as well, taking into account the parameters that make sense for a battery-powered device based on its battery capacity and intended battery life.